nutils/evaluable.py

```diff
@@ -2534,11 +2543,7 @@ def _simplified(self):
 
 class Real(Pointwise):
     __slots__ = ()
-
-    @types.apply_annotations
-    def __init__(self, arg: asarray):
-        assert arg.dtype == complex
-        super().__init__(arg)
+    return_type = lambda T: float if T in (float, complex) else int
```

Comment:
`numpy.real(numpy.array([False])).dtype` gives `dtype('bool')`

Reply:
changed to `return_type = lambda T: float if T == complex else T`